main.go

```diff
@@ -23,38 +27,46 @@ var postgresFunctions = []runtime.Transform{
 	},
 }
 
-func main() {
-	app := newApp()
-	err := app.Run(os.Args)
-	// If required flags aren't set, it will return with error before we could set up logging
-	if err != nil {
-		_, _ = fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
-	}
+var (
+	Network = "unix"
+	Address = "@crossplane/fn/default.sock"
+	// for testing purposes, especially on MacOS it's much easier to create local socket than whole directory structure and permissions
+	//Address = "default.sock"
+)
+
+type server struct {
+	pb.UnimplementedContainerizedFunctionRunnerServiceServer
 }
 
-func newApp() *cli.App {
-	app := &cli.App{
-		Name:    vp.AI.AppName,
-		Usage:   vp.AI.AppLongName,
-		Version: fmt.Sprintf("%s, revision=%s, date=%s", vp.AI.Version, vp.AI.Commit, vp.AI.Date),
-		Action:  run,
-		Flags: []cli.Flag{
-			runtime.NewLogLevelFlag(),
-			runtime.NewLogFormatFlag(),
-			runtime.NewFunctionFlag(),
-		},
+func (s *server) RunFunction(ctx context.Context, in *pb.RunFunctionRequest) (*pb.RunFunctionResponse, error) {
+	switch in.Image {
```

Comment:
Maybe use a map instead of switch? In the map you would have the name of the function-io and in value, the corresponding transformation go functions. After that you just do a simple get value by key to run the `RunCommand()`

Reply:
it won't solve anything - won't do

Reply:
it still makes no sense, because it tries to process the data and I need to put 
```
if _, exists := images[in.Image]; !exists {
   return earlier
}
```
no advantage for me

Reply:
It passes an empty array of functions, it will return the same iof right after the nested function has ended. You don't need that `if` clause.

Reply:
Sorry, resolving was my bad I miss-clicked because my browser was lagging and accidentally closed  :see_no_evil:

By 2 cent: I don't mind either way. With this few cases as switch is IMO fine. Once we have more function a map might be more readable. Right now it IMO doesn't make a difference

Reply:
It was a bit frustrating to unresolve the same issue twice, hence my updated comment :D. @wejdross seems we got a second opinion which does not mind to go with a switch though IMO we can improve already from now and read less code as we already have many many lines of code. Anyway if you feel there's no advantage you can resolve this discussion.